dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFile_V2.cpp

```diff
@@ -42,7 +43,10 @@ inline ColumnFilePersisteds transform_V2_to_V3(const ColumnFileV2s & column_file
         if (f->isDeleteRange())
             f_v3 = std::make_shared<ColumnFileDeleteRange>(std::move(f->delete_range));
         else
-            f_v3 = std::make_shared<ColumnFileTiny>(f->schema, f->rows, f->bytes, f->data_page_id);
+        {
+            auto schema = std::make_shared<ColumnFileSchema>(*(f->schema));
```

Comment:
It doesn't seems right: here you are not creating a managed ColumnFileSchema from the `SharedBlockSchemas`, but creating a wild one. So, it cannot reuse the same memory of `SharedBlockSchemas` I guess?

Reply:
Yes, I make a mistake here :sob: